### PR TITLE
Update nix flake to allow for user installations and add code examples

### PR DESCRIPTION
**Pull request by HPsaucii**
Pretty much what it says on the tin. Overhauls the nix flake to allow uses to properly install, and creates a section in the readme with instructions

### CODE REVIEW

**Review by nixos-goddess — APPROVED** [review]

**Review by electron271 — DISMISSED** [review]
checks fail

**Comment by nixos-goddess** [comment]
It's because of the workflow run

The build command in the workflow is set to `nix build .#magmawm`, but the flake was updated so that it doesn't have that attribute. Now the `.#magmawm` part needs to be `.#default`

**Review by nixos-goddess — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -72,6 +72,83 @@ The binary will be created in `./target/release/magmawm`.
 ## Install
 **MagmaWM** is still under heavy development and installation is not recommended.
 If you really want to, run the following command to install MagmaWM: 
+
+#### Nixos
+The following are two different ways to go about installing MagmaWM
+##### With overlays
+The cleaner option, but can cause issues with hash mismatching
+```nix
```
Comment (electron271) [inline README.md:80]: idealy these should be markdown dropdown menus (see https://gist.github.com/citrusui/07978f14b11adada364ff901e27c7f61)

**Review by electron271 — COMMENTED** [review]
idealy this should be changed

**Comment by mfdorst** [comment]
This should not have been merged. It breaks the dev shell due to a syntax error (extra semicolon at the end of `shell.nix`).

**Pull request merged**